src/main/java/io/excaliburfrc/robot/subsystems/Drive.java

```diff
@@ -19,15 +43,105 @@
   private final CANSparkMax rightFollower =
       new CANSparkMax(DrivetrainConstants.RIGHT_FOLLOWER_ID, MotorType.kBrushless);
 
+  private final Encoder rightEncoder =
+      new Encoder(
+          DrivetrainConstants.RIGHT_ENCODER_PORT_A, DrivetrainConstants.RIGHT_ENCODER_PORT_B);
+  private final Encoder leftEncoder =
+      new Encoder(DrivetrainConstants.LEFT_ENCODER_PORT_A, DrivetrainConstants.LEFT_ENCODER_PORT_B);
+
   private final DifferentialDrive drive = new DifferentialDrive(leftLeader, rightLeader);
 
+  private DifferentialDriveVoltageConstraint autoVoltageConstraint =
+      new DifferentialDriveVoltageConstraint(
+          new SimpleMotorFeedforward(kS, DrivetrainConstants.kV, kA),
+          DrivetrainConstants.DRIVE_KINEMATICS,
+          10);
+  TrajectoryConfig config =
+      new TrajectoryConfig(
+              DrivetrainConstants.MAX_SPEED_METERS_PER_SECOND,
+              DrivetrainConstants.MAX_ACCELERATION_METERS_PER_SECOND_SQUARED)
+          .setKinematics(DrivetrainConstants.DRIVE_KINEMATICS)
+          .addConstraint(autoVoltageConstraint);
+  Trajectory trajectory =
+      TrajectoryGenerator.generateTrajectory(
+          new Pose2d(0, 0, new Rotation2d(0)),
+          List.of(new Translation2d(1, 1), new Translation2d(2, -1)),
+          new Pose2d(3, 0, new Rotation2d(0)),
+          config);
+
+  private final DifferentialDriveOdometry odometry;
+  private final Gyro gyro = new ADXRS450_Gyro();
```

Comment:
We're using a NavX, which means using the `AHRS` class.

Reply:
You said it's the same thing...

Reply:
You can use it the same way (ie polymorphism), but it's a different sensor and class.